setup/mysql.sql

```diff
@@ -198,12 +205,18 @@ ALTER TABLE `domains` ENABLE KEYS;
 UNLOCK TABLES;
 
 --
+-- Generate password for siteadmin
+--
+
+SET @SITE_ADMIN_PW = SUBSTRING(TO_BASE64(LOWER(CONV(FLOOR(RAND() * 99999999999999), 10, 36))),1,10);
+
+--
 -- Seed the `users` table with the siteadmin user
 --
 
 LOCK TABLES `users` WRITE;
 ALTER TABLE `users` DISABLE KEYS;
-INSERT INTO `users` VALUES (1,1,'siteadmin','siteadmin','$6$uR.EiB1dj5rrvwMF$Qh5LgdjOZavKXwhi9IF0Yuzu7qxsG.dLTTB8e./55ZRNfBuZVLnfUSOEXa0oWT6174myO.WYkOy83HYWAKPbK/',65535,65535,'','','site',1,0,0,0,0,0,0,0,1,NULL,NULL,0,0,0,'SiteAdmin',0,0,NULL,NULL);
+INSERT INTO `users` VALUES (1,1,'siteadmin','siteadmin',ENCRYPT(@SITE_ADMIN_PW,CONCAT(@PW_PREFIX,MD5(RAND()))),65535,65535,'','','site',1,0,0,0,0,0,0,0,1,NULL,NULL,0,0,0,'SiteAdmin',0,0,NULL,NULL);
```

Comment:
There are some issues with this:

1. I've again defined the password hash prefix at the top of the file, so we would probably have to keep the paragraph about changing it if necessary (or we could use an MD5 hash instead). If we went with the MD5 hash, we should probably add a call in Vexim to update the hash upon login if its prefix does not match what's in the settings. On the other hand, we don't *have* to fix this, I think this is quite secure already, and we encourage the user to change the password anyway.
2. The salt is not as long as it could be with SHA-512. To fix this, we could `CONCAT()` the results of several `MD5(RAND())` calls. I would also probably enclose these in a `TO_BASE64()` call, to have mixed-case salt.
3. This generator doesn't add any special characters into the password or the salt. It is possible to add these as well, but I think defining a huge MySQL function for just this one password is probably overkill.
4. `ENCRYPT()` is deprecated. Although this does not have immediate effect, we should not forget this.